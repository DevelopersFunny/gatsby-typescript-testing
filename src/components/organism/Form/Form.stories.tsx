import React from 'react'
import { ComponentStory, ComponentMeta } from '@storybook/react'
import { userEvent, within } from '@storybook/testing-library'
import { Form } from '.'

export default {
  title: 'Organism/Form',
  component: Form,
  parameters: {
    fileName: __filename,
  },
  argTypes: {
    onSubmit: { action: true },
  },
} as ComponentMeta<typeof Form>

const Template: ComponentStory<typeof Form> = args => <Form {...args} />

export const EmptyForm = Template.bind({})

export const FilledForm = Template.bind({})
FilledForm.play = async ({ args, canvasElement }) => {
  const canvas = within(canvasElement)
  await userEvent.type(canvas.getByTestId('firstName'), 'My name', {
    delay: 100,
  })
  await userEvent.type(canvas.getByTestId('email'), 'hi@example.com', {
    delay: 100,
  })
  await userEvent.type(canvas.getByTestId('password'), 'supersecret', {
    delay: 100,
  })
  // await userEvent.click(canvas.getByRole('button'))
  // await waitFor(() => expect(args.onSubmit).toHaveBeenCalled())
}
