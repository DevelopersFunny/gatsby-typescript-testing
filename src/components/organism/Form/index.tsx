import React from 'react'
import { Button } from '../../atoms/Button'
import { InputLabel } from '../../molecules/InputLabel'

export interface IForm {
  onSubmit: () => void
}

export const Form = ({ onSubmit }: IForm) => {
  return (
    <>
      <form onSubmit={onSubmit}>
        <div className="px-2 py-2">
          <InputLabel
            label="First Name"
            type="text"
            id="firstName"
            name="firstName"
          />
        </div>
        <div className="px-2 py-2">
          <InputLabel label="Email" type="email" id="email" name="email" />
        </div>
        <div className="px-2 py-2">
          <InputLabel
            label="Password"
            type="password"
            id="password"
            name="password"
          />
        </div>
        <div className="px-2 py-2">
          <Button variant="contained">Login</Button>
        </div>
      </form>
    </>
  )
}
