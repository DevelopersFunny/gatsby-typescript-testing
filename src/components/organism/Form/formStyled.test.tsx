import React from 'react'
import * as stories from './Form.stories'
import { composeStories } from '@storybook/testing-react'
import { render, screen } from '@testing-library/react'
import { createSerializer, matchers } from '@emotion/jest'

/**
 * Emotion matchers and snapshots
 */
expect.extend(matchers)
expect.addSnapshotSerializer(createSerializer())

/**
 * Stories render
 */
const { EmptyForm } = composeStories(stories)

describe('Input render with props', () => {
  it('should render input', () => {
    render(<EmptyForm />)
    const formElement = screen.getByTestId('firstName')
    expect(formElement).toMatchSnapshot()
  })
})
