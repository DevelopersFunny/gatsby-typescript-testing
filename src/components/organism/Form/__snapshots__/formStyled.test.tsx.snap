// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input render with props should render input 1`] = `
.emotion-0 {
  height: 2.5rem;
  width: 100%;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  padding-left: 2.25rem;
  padding-right: 0.75rem;
  display: block;
  --tw-text-opacity: 1;
  color: rgba(17, 24, 39, var(--tw-text-opacity));
  background-color: rgba(0, 0, 0, 0);
  border-bottom-width: 2px;
  --tw-border-opacity: 1;
  border-color: rgba(209, 213, 219, var(--tw-border-opacity));
}

.emotion-0::-webkit-input-placeholder {
  font-size: 1rem;
  line-height: 1.5rem;
  color: rgba(0, 0, 0, 0);
}

.emotion-0::-moz-placeholder {
  font-size: 1rem;
  line-height: 1.5rem;
  color: rgba(0, 0, 0, 0);
}

.emotion-0:-ms-input-placeholder {
  font-size: 1rem;
  line-height: 1.5rem;
  color: rgba(0, 0, 0, 0);
}

.emotion-0::placeholder {
  font-size: 1rem;
  line-height: 1.5rem;
  color: rgba(0, 0, 0, 0);
}

@media (min-width: 640px) {
  .emotion-0 {
    font-size: 0.875rem;
    line-height: 1.25rem;
  }
}

.emotion-0::-webkit-input-placeholder {
  font-style: italic;
}

.emotion-0::-moz-placeholder {
  font-style: italic;
}

.emotion-0:-ms-input-placeholder {
  font-style: italic;
}

.emotion-0::placeholder {
  font-style: italic;
}

.emotion-0:focus {
  outline: 2px solid transparent;
  outline-offset: 2px;
  --tw-border-opacity: 1;
  border-color: rgba(37, 99, 235, var(--tw-border-opacity));
}

.emotion-0 {
  height: 2.5rem;
  width: 100%;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  padding-left: 2.25rem;
  padding-right: 0.75rem;
  display: block;
  --tw-text-opacity: 1;
  color: rgba(17, 24, 39, var(--tw-text-opacity));
  background-color: rgba(0, 0, 0, 0);
  border-bottom-width: 2px;
  --tw-border-opacity: 1;
  border-color: rgba(209, 213, 219, var(--tw-border-opacity));
}

.emotion-0::-webkit-input-placeholder {
  font-size: 1rem;
  line-height: 1.5rem;
  color: rgba(0, 0, 0, 0);
}

.emotion-0::-moz-placeholder {
  font-size: 1rem;
  line-height: 1.5rem;
  color: rgba(0, 0, 0, 0);
}

.emotion-0:-ms-input-placeholder {
  font-size: 1rem;
  line-height: 1.5rem;
  color: rgba(0, 0, 0, 0);
}

.emotion-0::placeholder {
  font-size: 1rem;
  line-height: 1.5rem;
  color: rgba(0, 0, 0, 0);
}

@media (min-width: 640px) {
  .emotion-0 {
    font-size: 0.875rem;
    line-height: 1.25rem;
  }
}

.emotion-0::-webkit-input-placeholder {
  font-style: italic;
}

.emotion-0::-moz-placeholder {
  font-style: italic;
}

.emotion-0:-ms-input-placeholder {
  font-style: italic;
}

.emotion-0::placeholder {
  font-style: italic;
}

.emotion-0:focus {
  outline: 2px solid transparent;
  outline-offset: 2px;
  --tw-border-opacity: 1;
  border-color: rgba(37, 99, 235, var(--tw-border-opacity));
}

<input
  class="peer emotion-0"
  data-testid="firstName"
  id="firstName"
  name="firstName"
  placeholder="First Name"
  type="text"
/>
`;
